packages/core/src/browser/shell/tab-bar-decorator.ts

```diff
@@ -0,0 +1,187 @@
+/********************************************************************************
+ * Copyright (C) 2019 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable } from 'inversify';
+import { Event, Emitter, Disposable, DisposableCollection } from '../../common';
+import { Title, Widget } from '@phosphor/widgets';
+
+export interface TabBarDecorator {
+
+    /**
+     * The unique identifier of the tab bar decorator.
+     */
+    readonly id: string;
+
+    readonly onDidChangeDecorations: Event<(titles: Title<Widget>[]) => Map<string, TabBarDecoration.Data>>;
```

Comment:
I wonder can we have simpler API for implementers:
```
readonly onDidChange: Event<void>();
decorate(titles: Title<Widget>[]): Map<string, TabBarDecoration.Data>;
```

an implementer fires an even then something changed and only implement `decorate` function

Also what is the contract for a map, i.e. what is a key?

Reply:
I simplified the API and documented the fields in the latest push as you suggested. The map has the URI of the editor as the key.

Reply:
I simplified the API and documented the fields in the latest push as you suggested. The map has the URI of the editor as the key.